packages/undo/src/index.ts

```diff
@@ -61,12 +61,12 @@ export const withUndo =
     throwReatomError(isAction(anAtom) || !isAtom(anAtom), 'atom expected')
 
     if (!anAtom.undo) {
-      const { name } = anAtom.__reatom
+      const { name, initValue } = anAtom.__reatom
 
-      const historyAtom = (anAtom.historyAtom = atom<Array<AtomState<T>>>(
-        [],
+      const historyAtom = anAtom.historyAtom = atom<Array<AtomState<T>>>(
```

Comment:
There are now need to modify AtomProto, just move the init logic to the data source.

```ts
      const historyAtom = (anAtom.historyAtom = atom<Array<AtomState<T>>>(
        [],
        `${name}.Undo._historyAtom`,
      ))
      anAtom.pipe(
        withInit((ctx, init) => {
          const state = init(ctx)
          historyAtom(ctx, [state])
          return state
        }),
      )
```

Reply:
*There is no need to

Reply:
Yes, indeed. Fixed it